src/Symfony/Component/Mime/NamedAddress.php

```diff
@@ -39,4 +48,17 @@ public function toString(): string
     {
         return $this->getEncodedNamedAddress();
     }
+
+    public static function fromString(string $string): self
+    {
+        if (false === strpos($string, '<')) {
+            return new self($string, '');
+        }
+
+        if (!preg_match(self::FROM_STRING_PATTERN, $string, $matches)) {
+            throw new \RuntimeException(sprintf('Could not parse "%s" to a "NamedAddress" instance.', $string));
```

Comment:
I propose to use an InvalidArgumentException here.
Not sure it should be in the Mime namespace.

Reply:
There's one in Mime already, so changed it to that